File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        
        int max = -1;
        vector <bool> result;
        
        for (int x : candies)
            if (max < x)
                max = x;
        
        for (int x : candies){
            if (x + extraCandies >= max)
                result.push_back(true);
            else
                result.push_back(false);
        }
        
        return result;
    }
};

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.cs
public class Solution {
    public IList<bool> KidsWithCandies(int[] candies, int extraCandies) {

        var maxNum = candies.Max();
        IList<bool> res = new List<bool>();
        
        for (var i = 0; i < candies.Length; ++i)
            if (candies[i] + extraCandies >= maxNum)
                res.Add(true);
            else
                res.Add(false);
        

        return res;


    }
}